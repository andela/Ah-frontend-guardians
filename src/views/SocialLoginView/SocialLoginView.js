import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import SocialAuthLogin from '../../components/SocialLogin/SocialAuthLogin';
import googleLoginAction from '../../actions/socialActions/googleAction';
import facebookLoginAction from '../../actions/socialActions/facebookAction';

export class SocialAuthView extends Component {
  handleFacebookReponse = response => {
    const { facebookLoginAction } = this.props;
    if (response.accessToken) {
      facebookLoginAction(response.accessToken);
    }
  };

  handleGoogleSuccess = response => {
    const { googleLoginAction } = this.props;
    if (response.tokenId) {
      googleLoginAction(response.tokenId);
    }
  };

  handleGoogleFailure = response => {
    const { googleLoginAction } = this.props;
    googleLoginAction(' Something went wrong');
  };

  componentWillReceiveProps(nextprops) {
    const { socialAuthState } = nextprops;

    if (socialAuthState) {
      localStorage.setItem('token', socialAuthState.token);
      this.props.history.push('/');
    }
  }

  render() {
    return (
      <div>
        <SocialAuthLogin
          responseFacebook={this.handleFacebookReponse}
          googleResponseSuccess={this.handleGoogleSuccess}
          googleResponseFailure={this.handleGoogleFailure}
        />
      </div>
    );
  }
}

export const mapStateToProps = state => {
  return {
    socialAuthState: state.social.isAuthenticated
  };
};

export default withRouter(
  connect(
    mapStateToProps,
    { googleLoginAction, facebookLoginAction }
  )(SocialAuthView)
);
