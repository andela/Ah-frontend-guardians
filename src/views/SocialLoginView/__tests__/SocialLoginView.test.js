import React from 'react';
import { shallow } from 'enzyme';
import { SocialAuthView, mapStateToProps } from '../SocialLoginView';

const push = jest.fn();

let mockProps = {
  googleLoginAction: jest.fn(),
  facebookLoginAction: jest.fn(),
  payload: {
    token: 'usertoken',
    errors: {
      errors: 'Something went wrong'
    }
  },
  socialAuthState: {
    isAuthenticated: false,
    facebookLogin: false,
    googleLogin: false,
    payload: '',
    token: ''
  },
  social: {
    isAuthenticated: false
  },
  history: { push }
};

let wrapper;

describe('<SocialAuthView />', () => {
  beforeEach(() => {
    wrapper = shallow(<SocialAuthView {...mockProps} />);
  });

  it('should render component correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.length).toEqual(1);
  });

  it('should call handle google login action', () => {
    wrapper.instance().handleGoogleFailure({
      token: 'access_token'
    });
    expect(wrapper.instance().props.googleLoginAction).toBeCalled();
  });

  it('should call handle google login successfully', () => {
    wrapper.setProps({
      response: {
        tokenId: 'token from google'
      }
    });
    let response = wrapper.instance().props.response;
    wrapper.instance().handleGoogleSuccess(response);
    expect(response).toEqual({ tokenId: 'token from google' });
  });

  it('should call facebook login action', () => {
    wrapper.setProps({
      response: {
        accessToken: 'token from facebook'
      }
    });
    wrapper.instance().handleFacebookReponse(wrapper.instance().props.response);
    expect(wrapper.instance().props.facebookLoginAction).toBeCalled();
  });

});
