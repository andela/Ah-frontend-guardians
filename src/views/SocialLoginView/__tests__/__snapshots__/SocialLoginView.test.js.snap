// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialAuthView /> should render component correctly 1`] = `ShallowWrapper {}`;
